v03_pipeline/lib/misc/io.py

```diff
@@ -148,7 +140,7 @@ def import_callset(
 def select_relevant_fields(
     mt: hl.MatrixTable,
     dataset_type: DatasetType,
-    additional_row_fields: None | list[str] = None,
+    additional_row_fields: None | dict[str, hl.expr.types.HailType | set] = None,
```

Comment:
These type hints are just for us right? I remember you said something about why we can't use mypy but can you remind me

Reply:
yeah... they're a best attempt rn (I'm pretty sure this type isn't even fully correct).

the issues with `mypy` were mostly with `luigi` and `hail`.  we have way too many cases where functions return something like `luigi.Target(x).exists()` or `hl.eval(hl.is_defined(ht.a))` where we know the type is `bool`, but neither library has the python typing integrations to make something like `mypy` work correctly.  There's definitely a potential path towards using the `mypy` non-strict mode and typing things as `Any`... I just opted against it because it's potentially even more confusing to have "bad" type checking rather than "no" type checking.